From 262ad0abcafa5dc96eec47aba22fbfbca0d177b1 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 1 Feb 2024 12:29:07 +0100
Subject: [PATCH 2/2] link libomp to compiler-rt on osx-arm

---
 openmp/runtime/src/CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/openmp/runtime/src/CMakeLists.txt b/openmp/runtime/src/CMakeLists.txt
index 619d4f7ba458..9ed52a43ae08 100644
--- a/openmp/runtime/src/CMakeLists.txt
+++ b/openmp/runtime/src/CMakeLists.txt
@@ -161,6 +161,11 @@ if(OPENMP_STANDALONE_BUILD OR (NOT OPENMP_ENABLE_LIBOMP_PROFILING))
   add_library(omp ${LIBOMP_LIBRARY_KIND} ${LIBOMP_SOURCE_FILES})
   # Linking command will include libraries in LIBOMP_CONFIGURED_LIBFLAGS
   target_link_libraries(omp ${LIBOMP_CONFIGURED_LIBFLAGS} ${LIBOMP_DL_LIBS})
+  if(APPLE AND CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
+    include(HandleCompilerRT)
+    find_compiler_rt_library(builtins CLANG_RT_BUILTINS_LIBRARY)
+    target_link_libraries(omp ${CLANG_RT_BUILTINS_LIBRARY})
+  endif()
 else()
   add_llvm_library(omp ${LIBOMP_LIBRARY_KIND} ${LIBOMP_SOURCE_FILES} PARTIAL_SOURCES_INTENDED
     LINK_LIBS ${LIBOMP_CONFIGURED_LIBFLAGS} ${LIBOMP_DL_LIBS}
